### PR TITLE
Pressable: show proper my-plan page

### PR DESCRIPTION
**Pull request by artpi**
This is a part of #8836.

- Introduces `state.sites.ID.options.pressable` option in sites redux tree
- Introduces `isPressableSite` selector
- Introduces `PRESSABLE_TRANSFER_SUCCESS` action to test pressable internals
- Disables domains manage link in my-plan page

## Testing

- Go to my-plan page for a business site
- Open redux dispatcher from redux devtools
- Dispatch `{ type: 'PRESSABLE_TRANSFER_SUCCESS', siteId: ID }`
- See that button disappears for managing domain OR whole call to action disappears for using domain credit

CC @rralian @gwwar @lamosty @retrofox @adambbecker

### CODE REVIEW

**Comment by rralian** [comment]
> How are you going to use your domain credit while on this plan? Do we implement domain management?

I've been discussing this a bit with @p3ob7o. No promises, but we may be able to enable domain management for these sites. In practice most users will not have a domain credit because automated-transfer will be limited to users with a custom domain. But it's possible for a business site to have a custom domain and still have an available domain credit, so we still need to work this out.

> Should we show "Advertising removed" section? Does not make sense for jetpack, but its still continuing features from previous plans

Yes. These are dotcom business site users who have made use of the custom theme/plugin functionality. Jetpack is an implementation detail that isn't relevant to the user.

> Is the .com theme catalogue going to be hooked up ? For jetpack sites we currently show only installed themes

Yes, that is the plan. See the spreadsheet which links back to the master thread for that effort.

> What about wordads instant activation? Is that going to be present in pressable?

It should, yes. There's work involved there too though.

File: client/state/sites/selectors.js [hunk 1]
```diff
@@ -134,6 +134,23 @@ export function isJetpackSite( state, siteId ) {
...
+ * @param  {Number}   siteId Site ID
+ * @return {?Boolean}        Whether site is a Pressable site
+ */
+export function isPressableSite( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	return site.options.pressable;
```
Comment (gwwar) [inline client/state/sites/selectors.js]: 👍 Thanks for adding this. Let's check in with @peterbutler as work starts on this.

File: client/blocks/product-purchase-features/product-purchase-features-list/custom-domain.jsx [hunk 2]
```diff
@@ -2,19 +2,17 @@
...
 
-export default ( { selectedSite, hasDomainCredit } ) => {
+export default ( props ) => {
 	return (
 		<div className="product-purchase-features-list__item">
-			<CustomDomainPurchaseDetail
-				selectedSite={ selectedSite }
-				hasDomainCredit={ hasDomainCredit }
-			/>
+			<CustomDomainPurchaseDetail { ...pick( props, [ 'selectedSite', 'hasDomainCredit', 'isPressableSite' ] ) } />
```
Comment (lamosty) [inline client/blocks/product-purchase-features/product-purchase-features-list/custom-domain.jsx]: Me too. I think we should pass all the props (with picking/omitting some) only if they are passed to a native component like `div` or `button`. Those component can accept various different props so it's better to simply pass down everything instead of listing the possible props.

However, in this case, we are passing down only three props so it's unnecessary to use this pattern IMO. :)

File: client/state/sites/selectors.js (hunk 1 above)
Comment (artpi) [inline client/state/sites/selectors.js]: I checked in with Peter and he is completely ok with that. Looks like pressable plan will be just a regular business plan.
I think this can be a good selector to make work on this easier.
Worse case scenario, we can remove it later, but even having something to grep can be a good start

File: client/state/sites/reducer.js [hunk 3]
```diff
@@ -75,6 +75,15 @@ export function items( state = {}, action ) {
 			}
 			return state;
 
+		case 'PRESSABLE_TEST':
+			const originalSite = state[ action.siteId ];
+			if ( originalSite ) {
+				return Object.assign( {}, state, {
+					[ action.siteId ]: merge( {}, originalSite, { jetpack: true, options: { pressable: true } } )
```
Comment (gwwar) [inline client/state/sites/reducer.js]: This is a good start, but we'll probably also want to mark when a site is in the process of transferring.

File: client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx [hunk 4]
```diff
@@ -29,10 +28,14 @@ const CustomDomainPurchaseDetail = ( { selectedSite, hasDomainCredit, translate
 				buttonText={ translate( 'Claim your free domain' ) }
 				href={ `/domains/add/${ selectedSite.slug }` }
 			/>
-			: null;
-
-	const renderHasCustomDomain = () =>
-		<PurchaseDetail
+		);
+	} else if ( ! hasDomainCredit && hasCustomDomain( selectedSite ) ) {
+		const actionButton = {};
```
Comment (gwwar) [inline client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx]: The name here makes it sound like this is a component instead of a grouping of props. Perhaps we should refactor PurchaseDetail a bit.

File: client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx [hunk 5]
```diff
@@ -41,31 +44,20 @@ const CustomDomainPurchaseDetail = ( { selectedSite, hasDomainCredit, translate
...
-			buttonText={ translate( 'Manage my domains' ) }
-			href={ `/domains/manage/${ selectedSite.slug }` }
-		/>;
-
-	const renderCustomDomainDetail = () => {
-		if ( hasCustomDomain( selectedSite ) ) {
-			return renderHasCustomDomain();
-		}
-
+			{ ...actionButton }
```
Comment (gwwar) [inline client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx]: I'm thinking it might make sense to update purchase detail to accept child components, and remove the PurchaseButton wrapper. (which only adds a class to button and is unnecessary ).

If this feels out of scope, lets mark this for another janitorial

File: client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx (hunk 4 above)
Comment (gwwar) [inline client/my-sites/upgrades/checkout-thank-you/custom-domain-purchase-detail.jsx]: A ternary might be a better fit, though the usage is still a bit odd.
```javascript
const purchaseDetailButtonProps = isPressableSite ? {} : {
    buttonText: translate( 'Manage my domains' ),
    href: `/domains/manage/${ selectedSite.slug }`
};
```

File: client/state/sites/reducer.js [hunk 6]
```diff
@@ -75,6 +75,15 @@ export function items( state = {}, action ) {
 			}
 			return state;
 
+		case 'PRESSABLE_TEST':
```
Comment (gwwar) [inline client/state/sites/reducer.js]: The name of this would more likely be something closer to `PRESSIBLE_TRANSFER_START`. Do you mind adding a new action type, even though we might rename this later.

**Comment by gwwar** [comment]
I left a few notes, but I think we can merge this in after creating a proper action type.

**Comment by artpi** [comment]
After @gwwar pointers, I:

- Introduced 2 selectors ( isPressableSite and isPressableSiteInTransfer )
- Introduced 2 actions:PRESSABLE_TRANSFER_SUCCESS  and PRESSABLE_TRANSFER_START
